src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java

```diff
@@ -860,6 +860,8 @@ protected void actionPerformed(final GuiButton btn) {
             final Enum next = Platform.rotateEnum(cv, backwards, iBtn.getSetting().getPossibleValues());
             if (btn == this.switchTallMode) {
                 tallMode = next == TerminalStyle.TALL;
+                AEConfig.instance.settings
+                        .putSetting(Settings.CRAFT_GUI_STYLE, tallMode ? TerminalStyle.TALL : TerminalStyle.SMALL);
```

Comment:
Save `Settings.TERMINAL_STYLE` setting here, also check terminals in AE2FC to add similar logic